flair/data.py

```diff
@@ -198,22 +200,76 @@ def __init__(self, text: str = None, use_tokenizer: bool = False, labels: List[L
 
         self._embeddings: Dict = {}
 
-        # optionally, directly instantiate with sentence tokens
+        # if text is passed, instantiate sentence with tokens (words)
         if text is not None:
 
-            # tokenize the text first if option selected, otherwise assumes whitespace tokenized text
+            # tokenize the text first if option selected
             if use_tokenizer:
-                sentences = split_single(text)
+
+                # use segtok for tokenization
                 tokens = []
+                sentences = split_single(text)
                 for sentence in sentences:
                     contractions = split_contractions(word_tokenizer(sentence))
                     tokens.extend(contractions)
 
-                text = ' '.join(tokens)
+                # determine offsets for whitespace_after field
+                index = text.index
+                running_offset = 0
+                last_word_offset = -2
+                last_token = None
+                for word in tokens:
+                    token = Token(word)
+                    self.add_token(token)
+                    try:
+                        word_offset = index(word, running_offset)
+                    except:
+                        word_offset = last_word_offset = +1
+                    if word_offset - 1 == last_word_offset:
+                        last_token.whitespace_after = False
```

Comment:
If we fail to determine the word_offset for the first word in the sentence, the `word_offset` would be -1 and the `last_word_offset` would be -2. In this case `word_offset - 1 == last_word_offset` would be true and you would try to set `whitespace_after` to False for `None`. This will throw an exception.

Reply:
Can't you set `last_word_offset` to -1 and check for `word_offset - 1 == last_word_offset and last_token is not None`?

Reply:
oh yes, you're right!